#!/usr/bin/env python3

PACKAGE = "offboard_py"  # Ensure this matches your package name

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# PID parameters for horizontal control
gen.add("pid_horizontal_Kp", double_t, 0, "Proportional Gain for Horizontal PID", 1.0, 0.0, 10.0)
gen.add("pid_horizontal_Ki", double_t, 0, "Integral Gain for Horizontal PID", 0.0, 0.0, 10.0)
gen.add("pid_horizontal_Kd", double_t, 0, "Derivative Gain for Horizontal PID", 0.0, 0.0, 10.0)

# PID parameters for vertical control
gen.add("pid_vertical_Kp", double_t, 0, "Proportional Gain for Vertical PID", 1.0, 0.0, 10.0)
gen.add("pid_vertical_Ki", double_t, 0, "Integral Gain for Vertical PID", 0.0, 0.0, 10.0)
gen.add("pid_vertical_Kd", double_t, 0, "Derivative Gain for Vertical PID", 0.0, 0.0, 10.0)

# PID parameters for distance control
gen.add("pid_distance_Kp", double_t, 0, "Proportional Gain for Distance PID", 1.0, 0.0, 10.0)
gen.add("pid_distance_Ki", double_t, 0, "Integral Gain for Distance PID", 0.0, 0.0, 10.0)
gen.add("pid_distance_Kd", double_t, 0, "Derivative Gain for Distance PID", 0.0, 0.0, 10.0)

# Follow distance parameter
gen.add("follow_distance", double_t, 0, "Follow Distance (m)", 10.0, -10.0, 100.0)

# Vision parameters
gen.add("virtual_plane", bool_t, 0, "Enable Virtual Plane", True)
gen.add("kalman_filter", bool_t, 0, "Enable Kalman Filter", True)
gen.add("IOU", double_t, 0, "Intersection over Union (IOU)", 1.0, 0.0, 1.0)
gen.add("accuracy", int_t, 0, "Accuracy of Vision System", 100, 0, 100)
gen.add("vertical_offset", double_t, 0, "Vertical Offset of Target (m)", 0.0, 0, 150.0)
gen.add("vision", bool_t, 0, "Vision Mode", False)

gen.add("intercept", bool_t, 0, "Intercept Mode", False)


exit(gen.generate(PACKAGE, "offboard_py", "VelocityConfig"))
